zsh/aliases.zsh

```diff
@@ -0,0 +1,60 @@
+# Aliases with existence checks for portability
+
+# Basic aliases that should work everywhere
+alias ls='ls --color=auto'
+
+# Conditional aliases based on available commands
+command_exists nvim && alias vim='nvim'
+
+# Diff alias - use GNU diff color if available
+[[ -n $(diff --version 2>/dev/null | grep GNU) ]] && alias diff='diff --color'
+
+# Date alias for macOS compatibility
+[[ ! -f /opt/homebrew/bin/gdate ]] && alias gdate='date'
+
+# Text viewing
+alias vat="vim -R -c 'set nomodifiable' -c 'nmap q :q!<CR>' -c 'set norelativenumber'"
+alias bat='echo "just use vat"'
+alias batp='echo "just use vat"'
+
+# Python-based aliases
+if command_exists python3; then
+    alias count_tokens='python3 -c '\''import tiktoken, sys; encoding = tiktoken.encoding_for_model("gpt-3.5-turbo"); token_count = len(encoding.encode(sys.stdin.read())); print(token_count)'\'''
+fi
+
+# Project-specific aliases
+[[ -f "$HOME/Projects/ai-asker/ask.py" ]] && alias ask="$HOME/Projects/ai-asker/ask.py"
+[[ -f "$HOME/Projects/ai-asker/ask.py" ]] && alias ask4="$HOME/Projects/ai-asker/ask.py --model gpt-4"
+[[ -f "$HOME/Projects/dotfiles/git/auto-commit.zsh" ]] && alias autocommit="$HOME/Projects/dotfiles/git/auto-commit.zsh"
+[[ -f "$DOTFILES_HOME/ghreadme.sh" ]] && alias ghreadme="$DOTFILES_HOME/ghreadme.sh"
```

Comment:
just making sure - will DOTFILES_HOME be set by now? I wonder if we should put some sort of assertion at the top of each file that the environment is how we expect